### PR TITLE
Remove dependency on Makfiles in endian_swapper

### PR DESCRIPTION
**Pull request by themperek**
Gives the ability to simplify Makefiles (remove pylib Makefiles).

Remove dependency on cocotb Makefiles in endian_swapper (test_endian_swapper_hal) example.
Cleanup.

### CODE REVIEW

File: examples/endian_swapper/cosim/Makefile [hunk 1]
```diff
@@ -1,30 +1,32 @@
...
 
-PYTHON_LIBDIR ?= /usr/lib64
-SWIG ?= swig
-CC=gcc
-ifeq ($(ARCH),i686)
-	CC += -m32
+PYTHON_LIBDIR := $(shell $(PYTHON_BIN) -c 'from distutils import sysconfig; print( sysconfig.get_config_var("LIBDIR") )')
+ifeq ($(PYTHON_LIBDIR),None)
+    PYTHON_LIBDIR := $(shell dirname $(PYTHON_BIN))/libs
 endif
```
Comment (imphil) [inline examples/endian_swapper/cosim/Makefile:6]: I think we have similar code in at least two places now. Can we add a --python-libdir and --python-includedir option to cocotb-config instead?

File: examples/endian_swapper/cosim/Makefile [hunk 2]
```diff
@@ -1,30 +1,32 @@
...
 
 .PHONY: all
 all: io_module.so _hal.so
 
 io_module.o: io.c io_module.h io.h
-	$(CC) $(GCC_FLAGS) -fPIC -pthread -fno-strict-aliasing -O2 -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 \
+	$(CC) $(CFLAGS) -fPIC -pthread -fno-strict-aliasing -O2 -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 \
 	-fexceptions --param=ssp-buffer-size=4 -D_GNU_SOURCE \
 	-fwrapv -DNDEBUG -I$(PYTHON_INCLUDEDIR) -c $< -o $@
```
Comment (imphil) [inline examples/endian_swapper/cosim/Makefile:23]: While you're at it: can you remove the NDEBUG (ideally in a separate commit)? It seems a dangerous choice for a piece of demo software.

File: examples/endian_swapper/cosim/Makefile [hunk 3]
```diff
@@ -1,30 +1,32 @@
...
 
 io_module.so: io_module.o
-	$(CC) -pthread -shared  $< -L$(PYTHON_LIBDIR) $(PYTHON_LD_FLAGS) -o $@
+	$(CC) $(CFLAGS) -pthread -shared  $< -L$(PYTHON_LIBDIR)  -o $@
 
 _hal.so: ../hal/endian_swapper_hal.c endian_swapper_hal_wrap.c io_module.so
-	$(CC) -g -ldl -shared -fPIC -I$(shell pwd) $(PYTHON_LD_FLAGS) -I$(PYTHON_INCLUDEDIR) -I../hal ../hal/endian_swapper_hal.c endian_swapper_hal_wrap.c io_module.so -o $@
+	$(CC) $(CFLAGS) -g -ldl -shared -fPIC -I$(shell pwd) -I$(PYTHON_INCLUDEDIR) -I../hal ../hal/endian_swapper_hal.c endian_swapper_hal_wrap.c io_module.so -o $@
 	if [ $(OS) = Darwin ]; then \
```
Comment (imphil) [inline examples/endian_swapper/cosim/Makefile:30]: That needs replacing as well, $(OS) isn't available any more.

File: examples/endian_swapper/cosim/Makefile (hunk 1 above)
Comment (themperek) [reply examples/endian_swapper/cosim/Makefile:6]: Only here in the makefiles (pylib will be removed). We will have only (some of it) in `_build_libs.py` as I can see.

Comment (themperek) [reply examples/endian_swapper/cosim/Makefile:6]: But this is can be done. Possible useful and much easier to control from Python any fixes.

Comment (eric-wieser) [reply examples/endian_swapper/cosim/Makefile:6]: I don't think there's much value in doing this. A better use of time would be for this example to show how to use setuptools to build an extension, and have it invoke `$(PYTHON_BIN) setup.py build_ext` or similar

Comment (themperek) [reply examples/endian_swapper/cosim/Makefile:6]: Long run I would like to do what @eric-wieser suggests.

Comment (imphil) [reply examples/endian_swapper/cosim/Makefile:6]: Don't worry if it's used only once.

File: examples/endian_swapper/cosim/Makefile (hunk 2 above)
Comment (themperek) [reply examples/endian_swapper/cosim/Makefile:25]: OK. Can do separate.

Comment (themperek) [reply examples/endian_swapper/cosim/Makefile:25]: https://github.com/cocotb/cocotb/pull/1621

Comment (eric-wieser) [reply examples/endian_swapper/cosim/Makefile:25]: Again, this might be easier to just let setuptools handle

**Comment by eric-wieser** [comment]
Note that gh-1585 has implicitly been resolved by doing this, as the code it refers to is now unused.

**Pull request merged**